play-json/src/main/scala/play/api/libs/json/JsConstraints.scala

```diff
@@ -173,6 +190,9 @@ trait PathWrites {
   def at[A](path: JsPath)(implicit wrs: Writes[A]): OWrites[A] =
     OWrites[A] { a => JsPath.createObj(path -> wrs.writes(a)) }
 
+  def nonDefault[A](path: JsPath, defaultValue: => A)(implicit wrs: Writes[A]): OWrites[A] =
```

Comment:
What is the use case? For now I don't see a sufficiently generic case where we would like a "default" value to be written as an empty object. Moreover if `wrs` is not an `OWrites` it would lead to a possible heterogeneous JSON format (a plain value if not default, an object otherwise...). At least that must be discussed out of the scope of this PR I think.